src/pgduckdb_ruleutils.cpp

```diff
@@ -27,14 +28,19 @@ extern "C" {
 
 extern "C" {
 char *
-pgduckdb_function_name(Oid function_oid) {
+pgduckdb_function_name_cpp(Oid function_oid) {
 	if (!pgduckdb::IsDuckdbOnlyFunction(function_oid)) {
 		return nullptr;
 	}
+
 	auto func_name = get_func_name(function_oid);
 	return psprintf("system.main.%s", quote_identifier(func_name));
 }
 
+char *
+pgduckdb_function_name(Oid function_oid) {
+	return InvokeCPPFunc(pgduckdb_function_name_cpp, function_oid);
```

Comment:
Why is this change needed? it seems like that the function doesn't actually run any C++ code.

Reply:
Ah right - it calls `pgduckdb::IsDuckdbOnlyFunction` which in itself isn't a C++-exception throwing function